src/cache.rs

```diff
@@ -179,16 +224,29 @@ impl Cache {
             })
             .collect();
 
-        // Try to find a platform specific path first.
+        // Get Custom Pages dir
+        let custom_dir = &config.directories.custom_pages_dir;
+
+        // Look up custom page (<name>.page). If it exists, return it directly
+        if let Some(custom_page) = custom_dir.as_ref().map(|dir| dir.join(custom_filename)) {
+            if custom_page.exists() && custom_page.is_file() {
+                return Some(PageLookupResult::with_page(custom_page));
+            }
+        }
+
+        let maybe_patch = Self::find_patch(&patch_filename, custom_dir.as_deref());
+
+        // Try to find a platform specific path next, append custom patch to it.
         if let Some(pf) = self.get_platform_dir() {
-            let pf_path = Self::find_page_for_platform(&page_filename, &cache_dir, pf, &lang_dirs);
-            if pf_path.is_some() {
-                return pf_path;
+            let page = Self::find_page_for_platform(&page_filename, &cache_dir, pf, &lang_dirs);
+            if page.is_some() {
+                return page.map(|lookup_result| lookup_result.with_optional_patch(maybe_patch));
```

Comment:
If the return value of `find_page_for_platform` would be `PathBuf` again, I would discard this shortcut with `is_some` and properly unpack the path and wrap it in `Some` again. This is mainly because I think it's strange to call `map` on an option that is knowingly `Some`. It could roughly look like this
```rust
let page = Self::find_page_for_platform(&page_filename, &cache_dir, pf, &lang_dirs);
if let Some(page_path) = page {
    return Some(LookupResult::with_page(page_path).with_optional_patch(maybe_patch));
}
```